src/common/ipc/update-available/index.ts

```diff
@@ -0,0 +1,48 @@
+import { UpdateInfo } from "electron-updater";
+
+export const UpdateAvailableChannel = "update-available";
+export const AutoUpdateLogPrefix = "[UPDATE-CHECKER]";
+
+/**
+ * [<back-channel>, <update-info>]
+ */
+export type UpdateAvailableFromMain = [string, UpdateInfo];
+
+export function areArgsUpdateAvailableFromMain(args: unknown[]): args is UpdateAvailableFromMain {
+  if (args.length !== 2) {
```

Comment:
Probably need to export to separate function because there's no clear description of this and following conditions. For example, why we return `false` if there's less or more 2 arguments? Separating into function with meaningful name will show the condition intent.

Reply:
I completely disagree the function name and signature explains exactly what this function is trying to do.

IPC calls can happen from anywhere so the handler that is called gets some array of elements (namely `unknown[]`). I use `unknown` here instead of `any` because of the difference of the coercion rules. Plus typescipt now recommends to use `unknown` where you don't know the types.

Finally, `UpdateAvailableFromMain` is by defenition a tuple (or an array of length 2). So if the arguments passed to the fuction are not of length 2 we know that they are invalid and thus `!(args is UpdateAvailableFromMain)`.